Blocker : Nokogumbo gem does not gem install (fail to link with Nokoguri)

Hello,

Tested on various linux distributions.

Here is the symptom on Mageia :

```
$ gem install nokogumbo -v '2.0.5' --source 'https://rubygems.org/' -- --with-ldflags="-lxml2"                  ERROR:  Could not find a valid gem 'nokogumbo' (= 2.0.5) in any repository
ERROR:  Possible alternatives: nokogumbo
[git@builder gitlab]$ cd
[git@builder ~]$ gem install nokogumbo -v '2.0.5' --source 'https://rubygems.org/' -- --with-ldflags="-lxml2" 
ERROR:  Could not find a valid gem 'nokogumbo' (= 2.0.5) in any repository
ERROR:  Possible alternatives: nokogumbo
[git@builder ~]$ RUBYOPT='--disable-did_you_mean' gem install nokogumbo -v '2.0.5' --source 'https://rubygems.org/' -- --with-ldflags="-lxml2" 
ERROR:  Could not find a valid gem 'nokogumbo' (= 2.0.5) in any repository
^CERROR:  Interrupted
[git@builder ~]$ gem install nokogumbo -v '2.0.4' --source 'https://rubygems.org/' -- --with-ldflags="-lxml2"                       Building native extensions with: '--with-ldflags=-lxml2'
This could take a while...
ERROR:  Error installing nokogumbo:
        ERROR: Failed to build gem native extension.

    current directory: /home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo
/usr/bin/ruby -I /usr/share/rubygems -r ./siteconf20210225-120197-1niegyi.rb extconf.rb --with-ldflags\=-lxml2
checking for libxml/tree.h in /home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2... yes
checking for nokogiri.h in /home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri... yes
creating Makefile

current directory: /home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo
make "DESTDIR=" clean
rm -f 
rm -f nokogumbo.so  *.o  *.bak mkmf.log .*.time

current directory: /home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo
make "DESTDIR="
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o nokogumbo.o -c nokogumbo.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o char_ref.o -c ./../../gumbo-parser/src/char_ref.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o tag.o -c ./../../gumbo-parser/src/tag.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o tokenizer.o -c ./../../gumbo-parser/src/tokenizer.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o util.o -c ./../../gumbo-parser/src/util.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o string_piece.o -c ./../../gumbo-parser/src/string_piece.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o svg_attrs.o -c ./../../gumbo-parser/src/svg_attrs.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o utf8.o -c ./../../gumbo-parser/src/utf8.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o error.o -c ./../../gumbo-parser/src/error.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o svg_tags.o -c ./../../gumbo-parser/src/svg_tags.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o vector.o -c ./../../gumbo-parser/src/vector.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o ascii.o -c ./../../gumbo-parser/src/ascii.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o attribute.o -c ./../../gumbo-parser/src/attribute.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o parser.o -c ./../../gumbo-parser/src/parser.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o tag_lookup.o -c ./../../gumbo-parser/src/tag_lookup.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o foreign_attrs.o -c ./../../gumbo-parser/src/foreign_attrs.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o string_buffer.o -c ./../../gumbo-parser/src/string_buffer.c
x86_64-mageia-linux-gnu-gcc -I. -I/usr/include -I/usr/include/ruby/backward -I/usr/include -I. -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri/include/libxml2 -I/home/git/.gem/ruby/gems/nokogiri-1.11.1-x86_64-linux/ext/nokogiri -I./../../gumbo-parser/src   -fPIC -O2 -g -pipe -Wformat -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector --param=ssp-buffer-size=4 -fasynchronous-unwind-tables -fPIC -std=c99 -DNGLIB=1  -o token_buffer.o -c ./../../gumbo-parser/src/token_buffer.c
rm -f nokogumbo.so
x86_64-mageia-linux-gnu-gcc -shared -o nokogumbo.so nokogumbo.o char_ref.o tag.o tokenizer.o util.o string_piece.o svg_attrs.o utf8.o error.o svg_tags.o vector.o ascii.o attribute.o parser.o tag_lookup.o foreign_attrs.o string_buffer.o token_buffer.o -L. -L/usr/lib64 -lxml2 -Wl,--as-needed -Wl,--no-undefined -Wl,-z,relro -Wl,-O1 -Wl,--build-id -Wl,--enable-new-dtags    -lruby  -lm   -lc
/usr/bin/ld: nokogumbo.o: in function `add_errors':
/home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo/nokogumbo.c:457: undefined reference to `cNokogiriXmlSyntaxError'
/usr/bin/ld: nokogumbo.o: in function `parse_continue':
/home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo/nokogumbo.c:551: undefined reference to `Nokogiri_wrap_xml_document'
/usr/bin/ld: nokogumbo.o: in function `Init_nokogumbo':
/home/git/.gem/ruby/gems/nokogumbo-2.0.4/ext/nokogumbo/nokogumbo.c:774: undefined reference to `mNokogiri'
collect2: error: ld returned 1 exit status
make: *** [Makefile:263: nokogumbo.so] Error 1

make failed, exit code 2

Gem files will remain installed in /home/git/.gem/ruby/gems/nokogumbo-2.0.4 for inspection.
Results logged to /home/git/.gem/ruby/extensions/x86_64-linux/2.7.0/nokogumbo-2.0.4/gem_make.out
```

It seems that the link with nokoguri is not made no matter if nokoguriwas installed with the system installer or with gem install

Tested and failed with nokogumbo v 2.0.2, 2.0.3, 2.0.4 on ruby 2.7.2 + gem 3.1.4 and ruby 2.5.8  + gem 2.6.14

All failed

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -13,6 +13,11 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Fixed
 ### Security
 
+## next
+### Fixed
+- Support Mageia distros when libxml2/libxslt system libraries are install. #165 (Thank you,
+  @pterjan!)
+
 ## [2.0.4] - 2020-11-27
 ### Fixed
 - Fixed a bug where `Nokogiri::HTML5.fragment(nil)` would raise an error. Now

--- a/ext/nokogumbo/extconf.rb
+++ b/ext/nokogumbo/extconf.rb
@@ -5,6 +5,7 @@ require 'nokogiri'
 
 $CFLAGS += " -std=c99"
 $LDFLAGS.gsub!('-Wl,--no-undefined', '')
+$DLDFLAGS.gsub!('-Wl,--no-undefined', '')
 $warnflags = CONFIG['warnflags'] = '-Wall'
 
 NG_SPEC = Gem::Specification.find_by_name('nokogiri', "= #{Nokogiri::VERSION}")

